python/verticox/vantage6.py

```diff
@@ -0,0 +1,145 @@
+import time
+from typing import List
+
+import grpc
+import pandas as pd
+from vantage6.client import ContainerClient
+from vantage6.tools.util import info
+
+from verticox import datanode
+from verticox.aggregator import Aggregator
+from verticox.grpc.datanode_pb2_grpc import DataNodeStub
+
+PORT = 8888
+MAX_RETRIES = 20
+GRPC_OPTIONS = [('wait_for_ready', True)]
+SLEEP = 5
+DATANODE_TIMEOUT = 60 * 60
+DATA_LIMIT = 10
+DEFAULT_PRECISION = 1e-5
+
+
+def _limit_data(data):
+    return data.iloc[:5]
+
+
+def verticox(client: ContainerClient, data: pd.DataFrame, feature_columns: List[str],
+             event_times_column: str, event_happened_column: str, datanode_ids: List[int] = None,
+             precision: float = DEFAULT_PRECISION,
+             *_args, **_kwargs):
+    """
+    TODO: Describe precision parameter
+    Args:
+        client:
+        data:
+        feature_columns:
+        event_times_column:
+        event_happened_column:
+        datanode_ids:
+        precision: determines precision in multiple places in the optimization process
+        *_args:
+        **_kwargs:
+
+    Returns:
+
+    """
+    event_times = data[event_times_column].values
+    event_happened = data[event_happened_column]
+
+    if datanode_ids is None:
+        organizations = client.get_organizations_in_my_collaboration()
+        datanode_ids = [organization.get("id") for organization in organizations]
+
+    datanode_input = {
+        "method": "run_datanode",
+        "kwargs": {
+            'feature_columns': feature_columns,
+            'event_time_column': event_times_column,
+            'event_happened_column': event_happened_column
+        }
+    }
+
+    # create a new task for all organizations in the collaboration.
+    info("Dispatching node-tasks")
+    task = client.create_new_task(
+        input_=datanode_input,
+        organization_ids=datanode_ids
+    )
+
+    addresses = _get_algorithm_addresses(client, datanode_ids, task['id'])
+
+    stubs = []
+    # Create gRPC stubs
+    for a in addresses:
+        stubs.append(_get_stub(a))
+
+    aggregator = Aggregator(stubs, event_times, event_happened, precision=precision)
+    aggregator.fit()
+
+    info('Verticox algorithm complete')
+    info('Retrieving betas')
+    betas = aggregator.get_betas()
+
+    info('Killing datanodes')
+    aggregator.kill_all_datanodes()
+    return betas
+
+
+def _get_stub(a):
```

Comment:
Add documentation here. Also maybe there is a clearer var name than `a` that can be used here?

Reply:
I made it clearer